packages/server/src/lib/cache.test.ts

```diff
@@ -26,81 +36,82 @@ describe('cache', () => {
 
         expect(fn).toHaveBeenCalledTimes(1);
 
-        reset();
+        fetchData.clear();
     });
 
     it('sets new cache value and refreshes', async () => {
-        const fn = jest.fn().mockImplementation(() => Promise.resolve(true));
-        const [reset, fetchData] = cacheAsync(fn, 60, 'test3');
+        const fn = jest.fn().mockReturnValue(Promise.resolve(true));
+        const fetchData = cacheAsync<boolean>(fn, { ttlSec: 60 });
 
         await fetchData();
 
         expect(fn).toHaveBeenCalledTimes(1);
 
-        jest.runOnlyPendingTimers(); // fast-forward to first refresh
+        jest.runAllTimers(); // fast-forward to first refresh
+
+        await fetchData();
 
         expect(fn).toHaveBeenCalledTimes(2);
 
-        reset();
+        fetchData.clear();
     });
 
     it('retries if initial request fails', async () => {
-        // Use counter to ensure it fails on the 1st invocation only
-        let counter = 0;
-        const err = new Error('ERROR');
-        const fn = jest.fn().mockImplementation(() => {
-            counter++;
-            if (counter === 1) {
-                return Promise.reject(err);
-            } else {
-                return Promise.resolve(true);
-            }
-        });
-
-        const [reset, fetchData] = cacheAsync(fn, 60, 'test4');
-
-        await expect(fetchData()).rejects.toEqual(
-            new Error('Failed to make initial request for test4: Error: ERROR'),
-        );
+        const fn = jest
+            .fn()
+            .mockReturnValueOnce(Promise.reject(new Error('ERROR')))
+            .mockReturnValue(Promise.resolve(true));
 
-        expect(fn).toHaveBeenCalledTimes(1);
+        const fetchData = cacheAsync<boolean>(fn, { ttlSec: 60 });
 
-        jest.runOnlyPendingTimers(); // fast-forward to retry
+        await expect(fetchData()).rejects.toEqual(new Error('ERROR'));
 
-        expect(fn).toHaveBeenCalledTimes(2);
+        expect(fn).toHaveBeenCalledTimes(1);
+
+        jest.useRealTimers();
+        await sleep(0);
+        jest.useFakeTimers();
```

Comment:
There's probably a better way to do this. I think `memoizee` doesn't evict the rejected promise from cache until the next time round the event loop, so we need to yield. Without this, it still returns the rejected promise.